src/main/webapp/app/modules/canvas/visualization/util/visualization-utils.ts

```diff
@@ -0,0 +1,166 @@
+    .sort(function (a, b) {
+      return sortBySequenceNumber(a.fieldType, b.fieldType);
+    });
+
+  const measures = fields
+    .filter(function (item) {
+      return item.feature && item.feature.featureType === 'MEASURE';
+    })
+    .map(function (item) {
+      const newItem = {};
+      //  angular.copy(item, newItem);
+      //  newItem.feature.name = newItem.feature.name;
+      return newItem;
+    })
+    .sort(function (a, b) {
+      return sortBySequenceNumber(a.fieldType, b.fieldType);
+    });
+  return {
+    measures: measures,
+    dimensions: dimensions,
+  };
+};
+
+const getDimensionsAndMeasuresForNotification = fields => {
+  const dimensions = fields
+    .filter(function (item) {
+      return item.feature && item.feature.featureType === 'DIMENSION';
+    })
+    .map(function (item) {
+      const newItem = {};
+      //   angular.copy(item, newItem);
+      //.feature.name = newItem.feature.name;
+      return newItem;
+    })
+    .sort(function (a, b) {
+      return sortBySequenceNumber(a.fieldType, b.fieldType);
+    });
+
+  const measures = fields
+    .filter(function (item) {
+      return item.feature && item.feature.featureType === 'MEASURE';
+    })
+    .map(function (item) {
+      const newItem = {};
+      //  angular.copy(item, newItem);
+      // newItem.feature.name = newItem.feature.name;
+      return newItem;
+    })
+    .sort(function (a, b) {
+      return sortBySequenceNumber(a.fieldType, b.fieldType);
+    });
+  return {
+    measures: measures,
+    dimensions: dimensions,
+  };
+};
+
+const getPropertyValue = (properties, propertyName, orElse) => {
+  const _properties = properties.filter(function (item) {
+    return item.propertyType.name === propertyName;
+  });
+  const property = _properties[0];
+
+  if (!property) {
+    if (isFunction(orElse)) {
+      return orElse();
+    } else {
+      return orElse;
+    }
+  } else {
+    if (property.value && property.value.value) {
+      return property.value.value;
+    } else {
+      return property.value;
+    }
+  }
+};
+
+const setPropertyValue = (properties, propertyName, orElse) => {
+  const _properties = properties.filter(function (item) {
+    return item.propertyType.name === propertyName;
+  });
+  const property = _properties[0];
+  if (property) {
+    if (property.value && property.value.value) {
+      property.value.value = 0;
+    } else {
+      property.value = 0;
+    }
+  }
+};
+
+const getFieldPropertyValue = (field, propertyName, orElse) => {
+  return getPropertyValue(field.properties, propertyName, orElse);
+};
+
+/**
+ * Normalize data according to given range
+ *
+ * @param {any} values array of values
+ * @param {any} lowBoundry left side of interval
+ * @param {any} highBoundry right side of interval
+ * @param {any} x value to be normalized
+ * @returns normalized value to given interval
+ */
+const normalize = (values, lowBoundry, highBoundry, x) => {
+  const maxValue = values.reduce(function (a, b) {
```

Comment:
to get min and max value needs to be separate function

Reply:
done